app/api/v1/dependencies/event.py

```
@@ -0,0 +1,206 @@
+    )
+
+
+def get_verification_runner():
+    """Return a callable that runs tenant verification with a fresh session and configured limits (for background jobs)."""
+    async def run_verification_for_tenant(tenant_id: str) -> ChainVerificationResult:
+        gen = get_db()
+        try:
+            session = await gen.__anext__()
+        except StopAsyncIteration:
+            await gen.aclose()
+            raise RuntimeError("Failed to obtain database session")
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)